app/assets/javascripts/discourse/app/lib/sidebar/user/tags-section/base-tag-section-link.js

```diff
@@ -1,6 +1,28 @@
+let customTagSectionLinkPrefixIcons = {};
+
+export function registerCustomTagSectionLinkPrefixIcon({
+  tagName,
+  prefixValue,
+  prefixColor,
+}) {
+  customTagSectionLinkPrefixIcons[tagName] = {
+    prefixValue,
+    prefixColor,
+  };
+}
+
+export function resetCustomTagSectionLinkPrefixIcons() {
+  for (let key in customTagSectionLinkPrefixIcons) {
+    if (customTagSectionLinkPrefixIcons.hasOwnProperty(key)) {
+      delete customTagSectionLinkPrefixIcons[key];
+    }
+  }
+}
+
```

Comment:
Do you see an issue just doing `customTagSectionLinkPrefixIcons = {};` ?

Reply:
Yes it suffers from the same problem which we've seen in the past where simply re-assigning the variables prevents the old object from being GC-ed if there is a reference being held on to the old object.